src/exercises/Exercises.js

```diff
@@ -117,34 +223,36 @@ export default function Exercises({ api, articleID }) {
     <>{strings.wordSourceDefaultText}</>
   );
 
+  const CurrentExercise = exerciseSession[currentIndex].type;
   return (
     <s.ExercisesColumn>
-      <s.LittleMessageAbove>{strings.wordSourcePrefix} {wordSourceText}</s.LittleMessageAbove>
-      <ProgressBar index={currentIndex} total={NUMBER_OF_EXERCISES} />
+      <s.LittleMessageAbove>
+        {strings.wordSourcePrefix} {wordSourceText}
+      </s.LittleMessageAbove>
+      <ProgressBar index={currentIndex} total={exerciseSession.length} />
 
       <s.ExForm>
-        {currentIndex % 2 === 0 && (
-          <FindWordInContext
-            bookmarkToStudy={currentBookmarkToStudy}
-            correctAnswer={correctAnswer}
-            notifyIncorrectAnswer={incorrectAnswerNotification}
-            api={api}
-          />
-        )}
-        {currentIndex % 2 === 1 && (
-          <MultipleChoice
-            bookmarkToStudy={currentBookmarkToStudy}
-            correctAnswer={correctAnswer}
-            notifyIncorrectAnswer={incorrectAnswerNotification}
-            api={api}
-          />
-        )}
+        <CurrentExercise
+          bookmarkToStudy={currentBookmarkToStudy}
+          correctAnswer={correctAnswer}
+          notifyIncorrectAnswer={incorrectAnswerNotification}
+          api={api}
+          setExerciseType={setCurrentExerciseType}
```

Comment:
even w/o looking at the code, this seems a bit strange.
why do we pass the "setExerciseType" and not the "exerciseType" directly?